api/api/constants/privilege.py

```diff
@@ -0,0 +1,72 @@
+import typing
+from dataclasses import dataclass
+
+from rest_framework.request import Request
+
+
+ANONYMOUS: typing.Literal["anonymous"] = "anonymous"
+AUTHENTICATED: typing.Literal["authenticated"] = "authenticated"
+PRIVILEGED: typing.Literal["privileged"] = "privileged"
+
+Level = typing.Literal[ANONYMOUS, AUTHENTICATED, PRIVILEGED]
+LEVELS = typing.get_args(Level)
+
+
+_PRIVILEGES = {}
+
+
+@dataclass
+class Privilege:
+    """
+    Privileges granted to applications upon approved request to Openverse maintainers.
+
+    Maintainers review requests for increased privileges on a per-case basis.
+
+    Distinct from ``rate_limit_model`` which only affects access rates rather than privileges.
+    """
+
+    slug: str
+    anonymous: typing.Any
+    authenticated: typing.Any
+    privileged: typing.Any
+
+    def __post_init__(self):
+        _PRIVILEGES[self.slug] = self
+
+    def request_level(self, request: None | Request) -> tuple[Level, typing.Any]:
+        """Retrieve the level of any request in relation to the privilege."""
+        if request is None or request.auth is None:
+            return ANONYMOUS, self.anonymous
+
+        if self.slug in request.auth.application.privileges:
+            return PRIVILEGED, self.privileged
+
+        return AUTHENTICATED, self.authenticated
+
+    @classmethod
+    @property
+    def choices(cls):
+        return ((slug,) * 2 for slug in _PRIVILEGES)
+
+
+PAGE_SIZE = Privilege(
+    "page_size",
+    anonymous=20,
+    authenticated=50,
+    # Max out privileged page size at the maximum authenticated
+    # pagination depth, otherwise privileged page size limit can
+    # contradict pagination depth limit.
+    privileged=240,
+)
+
+PAGINATION_DEPTH = Privilege(
```

Comment:
I found this to be confusing, because by the variable name I expected it to be the number of pages and then for the result count to be calculated as PAGE_SIZE * PAGINATION_DEPTH in the `clamp` functions later. Alternatively, if the Privilege were named 'RESULT_COUNT` instead that might also be clearer?

Reply:
Sure! I struggled to name this particular one as well. "Result count" is fine by me :+1:

Reply:
I've gone with "query depth" to disambiguate from the result count of the query response. Does that help? I noticed we do use `max_result_count` in the tallying code, so I'm fine using it if query depth still isn't clear enough, just let me know.

Reply:
Done in https://github.com/WordPress/openverse/pull/4372/commits/8d64c03117994a8bfe721002207a818ba2fac787